system/BaseModel.php

```diff
@@ -1657,7 +1657,7 @@ protected function objectToArray($data, bool $onlyChanged = true, bool $recursiv
         $properties = $this->objectToRawArray($data, $onlyChanged, $recursive);
 
         // Convert any Time instances to appropriate $dateFormat
-        if ($properties) {
+        if ($properties !== null && $properties !== []) {
```

Comment:
```suggestion
        if ((array) $properties !== []) {
```

Reply:
Is `$properties` always an array?
It seems `$this->objectToRawArray()` never returns `null`.

Reply:
@kenjis it check on return type defined

```php
    protected function objectToRawArray($data, bool $onlyChanged = true, bool $recursive = false): ?array
    {
```

Reply:
@samsonasik The return type is wrong.

See the code. The `$properties` is always an array.
https://github.com/codeigniter4/CodeIgniter4/blob/e1c2f5d23b7234208f11b06504ed00119b48b3ac/system/BaseModel.php#L1685-L1705
https://github.com/codeigniter4/CodeIgniter4/blob/e1c2f5d23b7234208f11b06504ed00119b48b3ac/system/Entity/Entity.php#L209

Reply:
I sent PR to fix the return type to `4.5`. #7986

Reply:
I added assert is array https://github.com/codeigniter4/CodeIgniter4/pull/7951/commits/f35199121f00f24e9c056467910f030cd9e4c609 , can be removed on 4.5